packages/shorebird_cli/lib/src/code_push_client_wrapper.dart

```diff
@@ -189,18 +218,26 @@ Please create a release using "shorebird release" and try again.
     required Map<Arch, ArchMetadata> architectures,
     required String platform,
   }) async {
+    // TODO(bryanoltman): update this function to only make one call to
+    // getReleaseArtifacts.
     final releaseArtifacts = <Arch, ReleaseArtifact>{};
     final fetchReleaseArtifactProgress = logger.progress(
       'Fetching release artifacts',
     );
     for (final entry in architectures.entries) {
       try {
-        final releaseArtifact = await codePushClient.getReleaseArtifact(
+        final artifacts = await codePushClient.getReleaseArtifacts(
```

Comment:
can we call `getReleaseArtifact` here to avoid code duplication until the TODO is resolved?

Reply:
Same here re: the progress per artifact